libcxx/test/libcxx/containers/associative/map/initialize_iterator.pass.cpp

```diff
@@ -0,0 +1,26 @@
+//===----------------------------------------------------------------------===//
```

Comment:
This is effectively testing an extension since https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2013/n3644.pdf mentions value-initialized iterators and this extension makes it work with _default_ initialized iterators. Let's move it under `test/extensions` and adjust the test comment to mention that this is an extension.